edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/DockerStats.cs

```diff
@@ -8,6 +8,19 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Docker.Models
 
     public class DockerStats
     {
+        [JsonConstructor]
+        public DockerStats(string name, Dictionary<string, DiskIO[]> block_io_stats, DockerCpuStats cpu_stats, MemoryStats memory_stats, Dictionary<string, NetworkInfo> networks, int? num_processes, PidsStats pids_stats, DateTime? read)
+        {
+            this.Name = name ?? "error_name";
+            this.BlockIoStats = block_io_stats ?? new Dictionary<string, DiskIO[]>();
+            this.CpuStats = cpu_stats ?? new DockerCpuStats(null, null, null);
+            this.MemoryStats = memory_stats ?? new MemoryStats(null, null, null);
+            this.Networks = networks ?? new Dictionary<string, NetworkInfo>();
+            this.NumProcesses = num_processes ?? 0;
+            this.PidsStats = pids_stats ?? new PidsStats(null);
+            this.Read = read ?? DateTime.MinValue;
```

Comment:
What are your thoughts on finding a way to use defaults that don't convey meaning? I can think of solutions to this, but they seem hacky. We should discuss.

Reply:
Agree. Will wrap in options